Compiler issues with jasmine.d.ts

Hi Folks, thanks for all the good work on the definitely typed modules. I'm using the jasmine.d.ts installed via tsd. The following 2 lines reproduces the issues I'm having:

/// <reference path="../typings/tsd.d.ts"/>
import \* as jasmine from 'jasmine';

When compiled with tsc they generate:

error TS2307: Cannot find module 'jasmine'.

which I think this is due to not having inverted commas around the name jasmine in the line

declare module jasmine {

If this is fixed then the next issues are:

error TS2503: Cannot find namespace 'jasmine'.

coming from the use of jasmine.Matchers & jasmine.Spy earlier in the file. Perhaps someone more familiar with type definitions could take a look.

Thanks

--- a/types/jasmine/index.d.ts
+++ b/types/jasmine/index.d.ts
@@ -7,28 +7,96 @@
 
 // For ddescribe / iit use : https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/karma-jasmine/karma-jasmine.d.ts
 
+/**
+ * Create a group of specs (often called a suite).
+ * @param description Textual description of the group
+ * @param specDefinitions Function for Jasmine to invoke that will define inner suites a specs
+ */
 declare function describe(description: string, specDefinitions: () => void): void;
 declare function fdescribe(description: string, specDefinitions: () => void): void;
 declare function xdescribe(description: string, specDefinitions: () => void): void;
 
+/**
+ * Define a single spec. A spec should contain one or more expectations that test the state of the code.
+ * A spec whose expectations all succeed will be passing and a spec with any failures will fail.
+ * @param expectation Textual description of what this spec is checking
+ * @param assertion Function that contains the code of your test. If not provided the test will be pending.
+ * @param timeout Custom timeout for an async spec.
+ */
 declare function it(expectation: string, assertion?: (done: DoneFn) => void, timeout?: number): void;
+
+/**
+ * A focused it
+ * If suites or specs are focused, only those that are focused will be executed.
+ * @param expectation 
+ * @param assertion
+ * @param timeout
+ */
 declare function fit(expectation: string, assertion?: (done: DoneFn) => void, timeout?: number): void;
 declare function xit(expectation: string, assertion?: (done: DoneFn) => void, timeout?: number): void;
 
-/** If you call the function pending anywhere in the spec body, no matter the expectations, the spec will be marked pending. */
+/**
+ * Mark a spec as pending, expectation results will be ignored.
+ * If you call the function pending anywhere in the spec body, no matter the expectations, the spec will be marked pending.
+ * @param reason
+ * @returns {} 
+ */
 declare function pending(reason?: string): void;
 
+/**
+ * Run some shared setup before each of the specs in the describe in which it is called.
+ * @param action Function that contains the code to setup your specs.
+ * @param timeout Custom timeout for an async beforeEach.
+ */
 declare function beforeEach(action: (done: DoneFn) => void, timeout?: number): void;
+
+/**
+ * Run some shared teardown after each of the specs in the describe in which it is called.
+ * @param action Function that contains the code to teardown your specs.
+ * @param timeout Custom timeout for an async afterEach.
+ */
 declare function afterEach(action: (done: DoneFn) => void, timeout?: number): void;
 
+/**
+ * Run some shared setup once before all of the specs in the describe are run.
+ * Note: Be careful, sharing the setup from a beforeAll makes it easy to accidentally leak state between your specs so that they erroneously pass or fail.
+ * @param action Function that contains the code to setup your specs.
+ * @param timeout Custom timeout for an async beforeAll.
+ */
 declare function beforeAll(action: (done: DoneFn) => void, timeout?: number): void;
+
+/**
+ * Run some shared teardown once before all of the specs in the describe are run.
+ * Note: Be careful, sharing the teardown from a afterAll makes it easy to accidentally leak state between your specs so that they erroneously pass or fail.
+ * @param action Function that contains the code to teardown your specs.
+ * @param timeout Custom timeout for an async afterAll
+ */
 declare function afterAll(action: (done: DoneFn) => void, timeout?: number): void;
 
+/**
+ * Create an expectation for a spec.
+ * @param spy
+ */
 declare function expect(spy: Function): jasmine.Matchers<any>;
+
+/**
+ * Create an expectation for a spec.
+ * @param actual
+ */
 declare function expect<T>(actual: ArrayLike<T>): jasmine.ArrayLikeMatchers<T>;
+
+/**
+ * Create an expectation for a spec.
+ * @param actual Actual computed value to test expectations against.
+ */
 declare function expect<T>(actual: T): jasmine.Matchers<T>;
 
+/**
+ * Explicitly mark a spec as failed.
+ * @param e
+ */
 declare function fail(e?: any): void;
+
 /** Action method that should be called when the async work is complete */
 interface DoneFn extends Function {
     (): void;
@@ -37,8 +105,19 @@ interface DoneFn extends Function {
     fail: (message?: Error|string) => void;
 }
 
+/**
+ * Install a spy onto an existing object.
+ * @param object The object upon which to install the Spy
+ * @param method The name of the method to replace with a Spy.
+ */
 declare function spyOn<T>(object: T, method: keyof T): jasmine.Spy;
 
+/**
+ * Install a spy on a property onto an existing object.
+ * @param object The object upon which to install the Spy
+ * @param property The name of the property to replace with a Spy
+ * @param accessType The access type (get|set) of the property to Spy on.
+ */
 declare function spyOnProperty<T>(object: T, property: keyof T, accessType: string): jasmine.Spy;
 
 declare function runs(asyncMethod: Function): void;
@@ -321,7 +400,20 @@ declare namespace jasmine {
         isNot?: boolean;
         message(): any;
 
+        /**
+         * 
+         * @param expected the actual value to be === to the expected value.
+         * @param expectationFailOutput 
+         * @returns {} 
+         */
         toBe(expected: Expected<T>, expectationFailOutput?: any): boolean;
+
+        /**
+         *  
+         * @param expected the actual value to be equal to the expected, using deep equality comparison.
+         * @param expectationFailOutput 
+         * @returns {} 
+         */
         toEqual(expected: Expected<T>, expectationFailOutput?: any): boolean;
         toMatch(expected: string | RegExp, expectationFailOutput?: any): boolean;
         toBeDefined(expectationFailOutput?: any): boolean;
@@ -594,4 +686,40 @@ declare namespace jasmine {
     export var HtmlSpecFilter: HtmlSpecFilter;
     export var DEFAULT_TIMEOUT_INTERVAL: number;
     export var MAX_PRETTY_PRINT_DEPTH: number;
+}
+
+declare module "jasmine" {
+    class jasmine {
+        constructor(options: any);
+        jasmine: jasmine.Jasmine;
+        addMatchers(matchers: jasmine.CustomMatcherFactories): void;
+        addReporter(reporter: jasmine.Reporter): void;
+        addSpecFile(filePath: string): void;
+        addSpecFiles(files: string[]): void;
+        configureDefaultReporter(options: any, ...args: any[]): void;
+        execute(files: any, filterString: any): any;
+        exitCodeCompletion(passed: any): void;
+        loadConfig(config: any): void;
+        loadConfigFile(configFilePath: any): void;
+        loadHelpers(): void;
+        loadSpecs(): void;
+        onComplete(onCompleteCallback: () => void): void;
+        provideFallbackReporter(reporter: jasmine.Reporter): void;
+        randomizeTests(value?: any): boolean;
+        seed(value: any): void;
+        showColors(value: any): void;
+        stopSpecOnExpectationFailure(value: any): void;
+        static ConsoleReporter(): any;
+        env: jasmine.Env;
+        reportersCount: number;
+        completionReporter: jasmine.CustomReporter;
+        reporter: jasmine.CustomReporter;
+        coreVersion(): string;
+        showingColors: boolean;
+        projectBaseDir: string;
+        printDeprecation(): void;
+        specFiles: string[];
+        helperFiles: string[];
+    }
+    export = jasmine;
 }

